core/src/value/partial.rs

```diff
@@ -0,0 +1,873 @@
+//! Handling of partial precision of Date, Time and DateTime values.
+
+use chrono::{DateTime, Datelike, FixedOffset, NaiveDate, NaiveTime, TimeZone, Timelike};
+use snafu::{Backtrace, OptionExt, ResultExt, Snafu};
+use std::convert::{TryFrom, TryInto};
+use std::fmt;
+use std::ops::RangeInclusive;
+
+#[derive(Debug, Snafu)]
+#[non_exhaustive]
+pub enum Error {
+    #[snafu(display("Date is invalid."))]
+    InvalidDate { backtrace: Backtrace },
+    #[snafu(display("Time is invalid."))]
+    InvalidTime { backtrace: Backtrace },
+    #[snafu(display("DateTime is invalid."))]
+    InvalidDateTime { backtrace: Backtrace },
+    #[snafu(display("To combine a DicomDate with a DicomTime value, the DicomDate has to be precise. Precision is: '{:?}'.", value))]
+    DateTimeFromPartials {
+        value: DateComponent,
+        backtrace: Backtrace,
+    },
+    #[snafu(display(
+        "'{:?}' has invalid value: '{}', must be in {:?}",
+        component,
+        value,
+        range
+    ))]
+    InvalidComponent {
+        component: DateComponent,
+        value: u32,
+        range: RangeInclusive<u32>,
+        backtrace: Backtrace,
+    },
+    #[snafu(display(
+        "Second fraction precision '{}' is out of range, must be in 0..=6.",
+        value
+    ))]
+    FractionPrecisionRange { value: u32, backtrace: Backtrace },
+    #[snafu(display(
+        "Number of digits in decimal representation of fraction '{}' does not match it's precision '{}'.",
+        fraction,
+        precision
+    ))]
+    FractionPrecisionMismatch {
+        fraction: u32,
+        precision: u32,
+        backtrace: Backtrace,
+    },
+    #[snafu(display("Conversion Failed."))]
+    Conversion { source: std::num::TryFromIntError },
```

Comment:
Could we expand this to also include which date/time component failed to convert?